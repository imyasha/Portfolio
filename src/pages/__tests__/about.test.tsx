import React from 'react';
import * as Gatsby from 'gatsby';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import About from '../about';

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      description: '',
      title: 'About',
      lang: 'en',
      meta: [],
      keywords: [],
    },
  },
}));

test('renders correctly', () => {
  const { container } = render(<About />);
  expect(container.firstChild).toMatchSnapshot();
});
