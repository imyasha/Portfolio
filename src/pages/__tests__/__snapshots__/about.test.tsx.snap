// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c11 {
  border-bottom: 1px solid;
  padding-bottom: 2rem;
  margin-bottom: 1rem;
  -webkit-break-inside: avoid;
  break-inside: avoid;
}

.c11 h1 {
  margin: 0 0 0.5rem;
}

.c11 h2 {
  font-weight: normal;
  font-size: 1rem;
  margin: 0 0 0.5rem;
}

.c15 {
  background: #fff;
  border: 2px solid #e94e1b;
  color: #e94e1b;
  padding: 0.5rem 2rem;
  border-radius: 99px;
  margin: 1rem 0;
  -webkit-transition: 0.2s;
  transition: 0.2s;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c15:hover {
  background: #e94e1b;
  color: #fff;
}

.c0 {
  width: 1140px;
  max-width: 94%;
  margin: 0 auto;
}

.c8 {
  background: transparent;
  border: none;
  margin-bottom: 1rem;
  opacity: 0.5;
}

.c8.active {
  opacity: 1;
  border-bottom: 2px solid;
}

.c1 {
  margin: 2rem 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c6 main {
  width: 70%;
  min-width: 280px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c6 aside {
  width: 30%;
  min-width: 280px;
}

.c6 aside ul {
  list-style: none;
  margin: 0;
  padding: 0;
}

.c6 aside ul li {
  margin-bottom: 0.5rem;
}

.c6 aside h2:first-of-type {
  margin: 0 0.5rem 1rem 0;
  display: inline-block;
  font-size: 1em;
}

.c9 {
  display: inline-block;
  padding: 0.25rem 0.5rem;
  margin: 0 0.5rem 0.5rem;
  border-radius: 99px;
  background: #333;
  color: #fff;
}

.c9.--technologies {
  background: #3e4153;
}

.c9.--frameworks {
  background: #1671cf;
}

.c9.--databases {
  background: #64cf16;
}

.c9.--cms {
  background: #e94e1b;
}

.c9.--design {
  background: #d03333;
}

.c2 {
  font-size: 3rem;
  margin: 0;
}

.c3 {
  font-size: 1.75rem;
  margin: 0;
}

.c4 {
  font-size: 1.2rem;
}

.c5 {
  font-size: 2.5rem;
}

.c12 {
  padding: 0 1rem;
  margin: 0;
  list-style: square;
}

.c13 {
  background: transparent;
  border: none;
  display: block;
  margin: 0 auto;
}

.c14 {
  text-align: center;
}

@media print {
  .c1 {
    border-bottom: 1pt solid #000;
    line-height: 1;
  }
}

@media print {
  .c9 {
    background: transparent;
    color: #000;
    border: 1px solid #000;
  }
}

@media print {
  .c2 {
    font-size: 22pt;
  }
}

@media print {
  .c3 {
    font-size: 16pt;
  }
}

@media print {
  .c4 {
    margin-top: 5pt;
    font-size: 11pt;
  }
}

@media print {
  .c5 {
    font-size: 22pt;
  }
}

@media print {
  .c10 {
    -webkit-column-count: 2;
    column-count: 2;
  }
}

@media print {
  .c7 {
    display: none;
  }
}

@media print {
  .c13 {
    display: none;
  }
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <h1
      class="c2"
      data-testid="headline"
    >
      Hey, I'm Jacob Herper
    </h1>
    <h2
      class="c3"
    >
      A senior Front-End Engineer from London, UK
    </h2>
    <p
      class="c4"
    >
      As a passionate front-end developer, I create amazing websites and web apps to make the internet a better place. I am an advocate for web performance and accessibility as well as a JAMstack enthusiast with experience in serverless technologies.
    </p>
    <p
      class="c4"
    >
      I am 
      29
       years old and have been a web developer for as long as I can think. The technologies I work with are JavaScript, HTML and CSS with a focus on the frameworks React.js, Gatsby, Next.js, Node and Express. I use code not only to do my day-to-day job, but also to solve everyday problems I come across.
    </p>
    <p
      class="c4"
    >
      When I am not writing code I love to spend time with my wife and
       
      2
       year old daughter at home in London or travelling around the world. We are quite a multi-cultural family with me having grown up in Germany ðŸ‡©ðŸ‡ª and my wife being from Mexico ðŸ‡²ðŸ‡½, which is why we raise our daughter trilingual. I myself speak five languages (some better than others). Furthermore I enjoy cooking fresh food when I come home after a long day at the office.
    </p>
  </section>
  <section
    class="c1"
  >
    <h2
      class="c5"
    >
      Skills
    </h2>
    <div
      class="c6"
    >
      <main>
        <div
          class="c7"
        >
          <div
            data-testid="filters"
          >
            Filter:
            <button
              class="c8"
              data-testid="filterTechnologies"
              name="technologies"
            >
              technologies
            </button>
            <button
              class="c8"
              data-testid="filterFrameworks"
              name="frameworks"
            >
              frameworks
            </button>
            <button
              class="c8"
              data-testid="filterDatabases"
              name="databases"
            >
              databases
            </button>
            <button
              class="c8"
              data-testid="filterCms"
              name="cms"
            >
              cms
            </button>
            <button
              class="c8"
              data-testid="filterVarious"
              name="various"
            >
              various
            </button>
            <button
              class="c8"
              data-testid="filterDesign"
              name="design"
            >
              design
            </button>
          </div>
        </div>
        <span
          class="c9 --technologies"
        >
          HTML5
        </span>
        <span
          class="c9 --technologies"
        >
          CSS
        </span>
        <span
          class="c9 --technologies"
        >
          Sass
        </span>
        <span
          class="c9 --technologies"
        >
          JavaScript
        </span>
        <span
          class="c9 --technologies"
        >
          TypeScript
        </span>
        <span
          class="c9 --technologies"
        >
          GraphQL
        </span>
        <span
          class="c9 --technologies"
        >
          ES6
        </span>
        <span
          class="c9 --technologies"
        >
          PHP
        </span>
        <span
          class="c9 --frameworks"
        >
          React
        </span>
        <span
          class="c9 --frameworks"
        >
          Vue.js
        </span>
        <span
          class="c9 --frameworks"
        >
          Gatsby
        </span>
        <span
          class="c9 --frameworks"
        >
          Next.js
        </span>
        <span
          class="c9 --frameworks"
        >
          Svelte
        </span>
        <span
          class="c9 --databases"
        >
          MySQL
        </span>
        <span
          class="c9 --databases"
        >
          MongoDB
        </span>
        <span
          class="c9 --databases"
        >
          Firebase
        </span>
        <span
          class="c9 --cms"
        >
          WordPress
        </span>
        <span
          class="c9 --cms"
        >
          Drupal
        </span>
        <span
          class="c9 --cms"
        >
          Contentful
        </span>
        <span
          class="c9 --cms"
        >
          Sanity.io
        </span>
        <span
          class="c9 --various"
        >
          SEO
        </span>
        <span
          class="c9 --various"
        >
          Scrum
        </span>
        <span
          class="c9 --various"
        >
          Jira
        </span>
        <span
          class="c9 --design"
        >
          Adobe Illustrator
        </span>
        <span
          class="c9 --design"
        >
          Adobe InDesign
        </span>
        <span
          class="c9 --design"
        >
          Adobe Photoshop
        </span>
        <span
          class="c9 --design"
        >
          Sketch
        </span>
        <span
          class="c9 --design"
        >
          Zeplin
        </span>
        <span
          class="c9 --design"
        >
          Figma
        </span>
        <h2>
          Soft skills
        </h2>
        <span
          class="c9 --softskill"
        >
          Strong communication
        </span>
        <span
          class="c9 --softskill"
        >
          Leadership
        </span>
        <span
          class="c9 --softskill"
        >
          Problem solving
        </span>
        <span
          class="c9 --softskill"
        >
          Creative thinking
        </span>
        <span
          class="c9 --softskill"
        >
          Flexible and adaptive
        </span>
        <span
          class="c9 --softskill"
        >
          Team player
        </span>
        <span
          class="c9 --softskill"
        >
          Always on time
        </span>
      </main>
      <aside>
        <h2>
          Languages
        </h2>
        <ul>
          <li>
            ðŸ‡¬ðŸ‡§ English (bilingual)
          </li>
          <li>
            ðŸ‡©ðŸ‡ª German (bilingual)
          </li>
          <li>
            ðŸ‡ªðŸ‡¸ Spanish (intermediate)
          </li>
          <li>
            ðŸ‡®ðŸ‡¹ Italian (basic)
          </li>
          <li>
            ðŸ‡·ðŸ‡º Russian (basic)
          </li>
        </ul>
      </aside>
    </div>
  </section>
  <section
    class="c1"
  >
    <h2
      class="c5"
    >
      Work Experience
    </h2>
    <div
      class="c10"
      id="jobs-closed"
    >
      <article
        class="c11"
        data-testid="jobDennisPublishing"
      >
        <h1>
          Senior Front-End Engineer
        </h1>
        <h2>
          <strong>
            <a
              href="https://www.dennis.co.uk/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Dennis Publishing
            </a>
          </strong>
           
          (in 
          London, UK
          )
        </h2>
        <ul
          class="c12"
        >
          <li>
            Joined the team at Dennis to help develop their internal web platform, built on React and Express
          </li>
        </ul>
        <small>
          10/2019
           to 
          present
        </small>
      </article>
      <article
        class="c11"
        data-testid="jobRaconteurMedia"
      >
        <h1>
          Front-End Developer
        </h1>
        <h2>
          <strong>
            <a
              href="https://www.raconteur.net/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Raconteur Media
            </a>
          </strong>
           
          (in 
          London, UK
          )
        </h2>
        <ul
          class="c12"
        >
          <li>
            Built and maintained internal websites using React (Next.js and Gatsby) with GraphQL and WordPress as a data source
          </li>
          <li>
            Extended internal front-end component library in React with Styled Components
          </li>
          <li>
            Built animated chart library for a large B2B client using ZingCharts, JavaScript and Gulp as a build tool
          </li>
        </ul>
        <small>
          01/2019
           to 
          10/2019
        </small>
      </article>
    </div>
    <button
      class="c13"
    >
      Show more
    </button>
  </section>
  <section
    class="c1"
  >
    <div
      class="c14"
    >
      <p
        class="c4"
      >
        Recruiters, you can find my up-to-date CV as a PDF download by clicking the button below.
      </p>
      <a
        class="c15"
        download="jacob_herper-cv.pdf"
        href="/docs/cv-jacob_herper-2020.pdf"
      >
        Download CV
      </a>
    </div>
  </section>
</div>
`;
